File: 14-Longest-Common-Prefix.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        strs.sort()
        st=strs[0]
        end=strs[len(strs)-1]
        s=""
        for i in range(0,min(len(st),len(end))):
            if st[i]==end[i]:
                s+=st[i]
            else:
                break
        return s
        

File: 14-Longest-Common-Prefix.java
class Solution {
    public String longestCommonPrefix(String[] str) {
        Arrays.sort(str);
        String st=str[0];
        String end=str[str.length-1];
        StringBuffer sb = new StringBuffer();
        for(int i=0; i<Math.min(st.length(),end.length()); i++){
            if(st.charAt(i) == end.charAt(i)) sb.append(st.charAt(i));
            else    break;
        }
        return sb.toString();
    }
}